@types/global.d.ts

```diff
@@ -0,0 +1,21 @@
+/* eslint-disable no-var */
+/* eslint-disable vars-on-top */
+
+declare global {
+	import mongooseInstance from 'mongoose'
+
+	interface MongooseGlobalVar {
+		conn: typeof mongooseInstance | null
+		promise: Promise<typeof mongooseInstance> | null
+	}
+
+	/**
+	 * `global.mongoose` is used to maintain only *ONE*
+	 * active connection to mongoose at a time, by caching it in a global variable.
+	 *
+	 * This prevents connections from growing exponentially during API Routes usage
+	 */
+	var mongoose: MongooseGlobalVar
```

Comment:
Any reason why you used `var` here instead of `let`?

Reply:
Actually, I tried using `let` but the type won't work that way. 
Here is the video:

https://user-images.githubusercontent.com/65458493/184605493-bd2a1f4b-9518-45b3-9694-6f8e5362ff56.mov

Reply:
Ok then.